pkg/generators/forwarding/fluentd/templates.go

```diff
@@ -291,24 +291,32 @@ const fluentConfTemplate = `{{- define "fluentConf" -}}
   </filter>
 
   # Relabel specific source tags to specific intermediary labels for copy processing
-{{ if .CollectInfraLogs }}
+  # Earlier matchers remove logs so they don't fall through to later ones.
```

Comment:
the bug is that we read infra logs when only "application" is configured
this section of fluent.conf routes the logs to proper destination type, but still logs from infra space will be collected.

perhaps we should specifically exclude infra logs when "application" is configured by generating something like:
```
path "/var/log/containers/*.log"
exclude_path ["/var/log/containers/*_default_*", "/var/log/containers/*_kube-*_*", "/var/log/containers/*_openshift-*_*", "/var/log/containers/"]
```
and for "infrastructure"
```
path ["/var/log/containers/*_default_*", "/var/log/containers/*_kube-*_*", "/var/log/containers/*_openshift-*_*", "/var/log/containers/"]
```
and if both "appliation" and "infrastructure"
```
path "/var/log/containers/*.log"
```
(with appropriate check to exclude logging logs)

Reply:
but if our approach is to capture ALL logs, and then decide where we want to send them (or not send them), then the solution is ok. but this conflicts with the solution for namespace filtering, which filters the logs at the log collection level.

Reply:
We were already scraping all the logs, these sections are simply re-labelling them to the right destinations. The problem was that if you don't ask for infra those, there was no infra match rule and they all fall thru to the applicaiton match rule.  My fix always creates all the sections but sends to 'null' instead of relabelling. It is no worse than the existing code and the same as far as namespaces are concerned.

I agree there is wasted work in scraping logs just to send them to 'null', we should review how we configure fluentd to address that. We can raise a new story for that.

Does that make sense or am I missing something?

Reply:
yes, it makes sense.

/lgtm

Reply:
@alanconway Does this PR make the note in the docs irrelevant?
_you must configure a pipeline and output for all three types of logs, application, infrastructure, and audit. If you do not specify a pipeline and output for a log type, those logs are not stored and will be lost._

Is @vimalk78 comment about needing pipelines for both infra and app required to get the container logs? Qialoing's  comment in the BZ  supports this: https://bugzilla.redhat.com/show_bug.cgi?id=1866531#c6

Reply:
@mburke5678 , the PR #699 fixes comments from Qiaolong.